### PR TITLE
methods-method_overloading-hard-q7

### PR DESCRIPTION
**Pull request by ketphan02**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [x] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/methods/method_overloading/hard/q7/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                Arguments.of(1.1, 1, 564, 382194, 0.1, 381630),
+                Arguments.of(67.32781, 467.237, 38, 2, 399.90919, 36)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("mainMethodInputProvider")
+    void correctMainMethodOutput(double x, double y, int a, int b, double ans1, int ans2) {
+        runWithInput(x + " " + y + " " + a + " " + b);
+        assertEquals(x, Double.parseDouble(getItemByName("d1")), 0.0000001, "Your program does not order the double values correctly.");
```
Comment (Hedgemon4) [inline src/methods/method_overloading/hard/q7/MainTest.java:85]: These four asserts won't be necessary based on how I suggested changing the question

File: src/methods/method_overloading/hard/q7/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    @ParameterizedTest
+    @MethodSource("doubleInputProvider")
+    void correctDoubleCalcDistanceMethod(double x, double y, double distance) throws Throwable {
+        Object[][] arguments = {
+                {x, double.class},
+                {y, double.class}
+        };
+        MethodTest m = new MethodTest(Distances.class, "calcDistance", arguments);
+        Object output = m.callMethod();
+        String errMsg = "Your calcDistance method does not return the correct distance for two doubles.";
```
Comment (Hedgemon4) [inline src/methods/method_overloading/hard/q7/MainTest.java:103]: The error message can be improved

File: src/methods/method_overloading/hard/q7/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    @ParameterizedTest
+    @MethodSource("integerInputProvider")
+    void correctIntegerCalcDistanceMethod(int x, int y, int distance) throws Throwable {
+        Object[][] arguments = {
+                {x, int.class},
+                {y, int.class}
+        };
+        MethodTest m = new MethodTest(Distances.class, "calcDistance", arguments);
+        Object output = m.callMethod();
+        String errMsg = "Your calcDistance method does not return the correct distance for two integers.";
```
Comment (Hedgemon4) [inline src/methods/method_overloading/hard/q7/MainTest.java:116]: Same thing here with the error message

File: src/methods/method_overloading/hard/q7/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        CustomAssertions._assertEquals(distance, output, errMsg);
+    }
+}
```
Comment (Hedgemon4) [inline src/methods/method_overloading/hard/q7/MainTest.java:119]: Add EOF

File: src/methods/method_overloading/hard/q7/Distances.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package methods.method_overloading.hard.q7;
+
+/*
+Write a Java Program that calculates the distance between 2 points given by the user. The first set of points are
```
Comment (Hedgemon4) [inline src/methods/method_overloading/hard/q7/Distances.java:4]: Do you think it might be a good idea to give the students one of the two points in each set so we can add variation on the website?

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]
